nuxeo-runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java

```diff
@@ -1668,7 +1669,7 @@ public void addTemplate(String templatesToAdd) throws ConfigurationException {
         String currentTemplatesStr = userConfig.getProperty(PARAM_TEMPLATES_NAME);
         List<String> templatesList = new ArrayList<>();
         templatesList.addAll(Arrays.asList(currentTemplatesStr.split(TEMPLATE_SEPARATOR)));
-        return templatesList;
+        return templatesList.stream().map(t -> replaceEnvironmentVariables(t)).collect(Collectors.toList());
```

Comment:
`t -> replaceEnvironmentVariables(t)` could be replaced by `this:: replaceEnvironmentVariables`.